### PR TITLE
Externalize SPARQL queries into templates

### PR DESCRIPTION
**Pull request by cthoyt**
Reference #791 (original issue) and references #906 (concrete solution proposal)

I haven't marked this PR as closing either issue because there are still many other HTML templates that would need to be modified, so consider this PR as a pilot that can be easily followed by one or more PRs to address the remaining HTML templates.

## Changes

- Move SPARQL strings in HTML templates into dedicated SPARQL templates
- Update README with explanation for new contributors on how to use templating

### CODE REVIEW

**Comment by egonw** [comment]
Nice!

**Comment by Daniel-Mietchen** [comment]
Looks promising.

**Comment by egonw** [comment]
@fnielsen, if you agree with this going forward, then I can resolve the conflicts.

**Comment by cthoyt** [comment]
@fnielsen @Daniel-Mietchen any feedback on this one?

**Comment by fnielsen** [comment]
I am wondering about the naming, e.g., author_topics.sparql. Does that mean the topics panel on the author aspect (yes, here) or could it mean a SPARQL query on an author topics subaspect. Maybe this is not a problmem.

**Comment by cthoyt** [comment]
@fnielsen good point. Maybe it would make more sense to have a hierarchical folder structure, where the folders were named by the aspect for which the enclosed files would be stored. I will make a small reorganization as an example

File: scholia/app/templates/sparql/author/publications.sparql [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#defaultView:Table
+SELECT
+  (MIN(?dates) AS ?date)
+  ?work ?workLabel
+  (GROUP_CONCAT(DISTINCT ?type_label; separator=", ") AS ?type)
+  (SAMPLE(?pages_) AS ?pages)
+  ?venue ?venueLabel
+  (GROUP_CONCAT(DISTINCT ?author_label; separator=", ") AS ?authors)
+  (CONCAT("../authors/", GROUP_CONCAT(DISTINCT SUBSTR(STR(?author), 32); separator=",")) AS ?authorsUrl)
```
Comment (cthoyt) [inline scholia/app/templates/sparql/author/publications.sparql]: Note that this line was the offending one that needed to be resolved. Now the updates that were originally in the HTML file are here :)

File: scholia/app/templates/sparql/author/publications.sparql [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+WHERE {
+  ?work wdt:P50 wd:{{ q }} .
+  ?work wdt:P50 ?author .
+  OPTIONAL {
+    ?author rdfs:label ?author_label_ . FILTER (LANG(?author_label_) = 'en')
+  }
+  BIND(COALESCE(?author_label_, SUBSTR(STR(?author), 32)) AS ?author_label)
```
Comment (cthoyt) [inline scholia/app/templates/sparql/author/publications.sparql]: @fnielsen @egonw any idea why the code highlighting stops working here? Is there a problem with the SPARQL syntax?

Comment (fnielsen) [reply scholia/app/templates/sparql/author/publications.sparql]: I cannot see any reason.

Comment (cthoyt) [reply scholia/app/templates/sparql/author/publications.sparql]: Okay. That's not really relevant for the PR, I was just curious

**Comment by egonw** [comment]
Draft patch by Finn: https://github.com/fnielsen/scholia/compare/feat/external-sparql-971

**Comment by egonw** [comment]
@Daniel-Mietchen, @fnielsen, I will extract the two patches that are not about externalisation.

**Comment by cthoyt** [comment]
@egonw is there renewed interest in this? Is there anything I can do to help?

**Comment by egonw** [comment]
We had not lost interest. But there were a lot of technical wishes :) the wished design now is ready and tomorrow I expect we discuss how to roll out out to all aspects. Your help is of course welcome. Right now, you could check the new patch, sketching the design.

**Comment by egonw** [comment]
I closed this PR as it is replaced by #1265, #1266, and https://github.com/fnielsen/scholia/compare/feat/external-sparql-971

**Pull request closed without merging**